internal/utils/sets/types.go

```diff
@@ -0,0 +1,49 @@
+package sets
+
+import (
+	"github.com/golang-collections/collections/set"
+	"sort"
+)
+
+type String struct {
+	set.Set
+}
+
+func NewString(values ...string) *String {
+	s := &String{
+		*set.New(),
+	}
+	for _, v := range values {
+		s.Insert(v)
+	}
+	return s
+}
```

Comment:
```
func NewString(values ...string) *String {
    return &String{
        *set.New(values),
    }
}
```

Reply:
This does not compile https://go.dev/play/p/iDXapPejjLy

Reply:
It compiles, but doesn't run :) I should've written
```
func NewString(values ...string) *String {
    return &String{
        *set.New(values...),
    }
}
```
Now that doesn't compile:
```
./prog.go:17:12: cannot use values (variable of type []string) as []interface{} value in argument to set.New
```
A reasonable person would expect that to compile, but this is Go, so https://go.dev/doc/faq#convert_slice_of_interface. I give up.